src/http-api.ts

```diff
@@ -288,6 +288,11 @@ export class HttpApi<S extends Schema> extends EventEmitter {
    */
   parseError(body: any) {
     const errors = body;
+
+    // XML response
+    if (errors['Errors']) {
+      return errors['Errors']['Error'];
+    }
```

Comment:
this cover errors in XML responses from the REST API, if you remove this block then the message in this test includes the whole XML instead of just the error props:
https://github.com/jsforce/jsforce/pull/1390/files#diff-de7739647545a012d3fb1a85782ad2a184f9b8803a7ffa84cedaf3774d58b82aR238

example using `org api`:
```
➜  jsforce git:(cd/http-module-tests) ✗ sf org api '/services/data/v59.0/sobjects/Broker__c/a008N0000032UmoQA' -o test-rkr5unx057se@example.com --header 'Accept:application/json'
[
  {
    "errorCode": "NOT_FOUND",
    "message": "Provided external ID field does not exist or is not accessible: a008N0000032UmoQA"
  }
]
➜  jsforce git:(cd/http-module-tests) ✗ sf org api '/services/data/v59.0/sobjects/Broker__c/a008N0000032UmoQA' -o test-rkr5unx057se@example.com --header 'Accept:application/xml'
<?xml version="1.0" encoding="UTF-8"?><Errors><Error><errorCode>NOT_FOUND</errorCode><message>Provided external ID field does not exist or is not accessible: a008N0000032UmoQA</message></Error></Errors>
```